### ISSUE TITLE
Using the Hand tool sometimes jumps the view to the newly selected page. 

### ISSUE THREAD

**Issue by pindakaas42** [xournalpp/xournalpp#2444]
Affects versions :

    Xubuntu 18.04
    Desktop: Xfce
    version of libgtk 3.22.30
    Version of Xournal++: 1.0.19
    Installation method: PPA

Describe the bug
Using the hand tool on a page that is not currently highlighted as active (active is highlighted with red borders), the view then jumps to the top of the page, that has been grabbed with the hand tool. This happens inconsistently, sometimes when a document is opened. But if this behaviour is seen it is consistent within the instance of xournalpp. 
This jumping is very disorienting and the normal behaviour is a lot nicer. 

To Reproduce
Steps to reproduce the behaviour:
Open a multi A4 page document.
Do a zoom to fit.
Using the hand tool on a page that is not currently highlighted as active (active is highlighted with red borders), the view then jumps to the top of the page, that has been grabbed with the hand tool.
This happens inconsistently, sometimes when a document is opened. But if this behaviour is seen it is consistent within taht instance of xournalpp. 

Expected behaviour:
The behaviour that normally happens, meaning the view does not jump to the top of the page but stays put.

**Comment by rolandlo** [comment]
This looks like it is related to Pull Request #2425. If you have the chance to try out the PR, please check if that bug is solved there.

**Comment by pindakaas42** [comment]
That is over my pay grade. Just wanted to report the issue.

**Comment by LittleHuba** [comment]
@pindakaas42 prepare for a small rant:

As this is a free opensource project with no monetary gain for any party we could argue that all this is above our pay grade as well.

Thus we ask you kindly to help us out with checking where the bug originates from and to provide us with further details. Not all bugs occur for all users on all devices, so we are sometimes unable to actually check what happens. On top comes that we are a small team that maintains a huge software used by many users, which all report bugs. We try our best to catch up with the growing list of bug report but somewhere we have our limits. 

Therefore just telling us this is above your pay grade does a) not help us in fixing the bug, b) make this bug way harder to fix as it would be otherwise and c) does not motivate us at all to give this bug any priority in the near future.

TLDR if you want this fixed, help us out by checking if PR #2425 fixes this bug or ignore all of this and wait for some other user to take this up and help us out or till we come around to fix this (in half a year earliest). Your turn.

**Comment by pindakaas42** [comment]
In case you have just not understood what I meant, I have no Idea how one would check out how Pull request number so and so impacts some bug I found. Could you have the grace to help me with that?

**Comment by rolandlo** [comment]
@pindakaas42 
Let me help you with how to test out a Pull Request (short: PR)
First you need to be able to build Xournal++ from source. For Linux this is described on [the Linux build instruction wiki page](https://github.com/xournalpp/xournalpp/blob/master/readme/LinuxBuild.md). 
In short: you need to install the required dependencies, clone the repository, compile with the `cmake` tool and run from the `build` directory via the terminal command `src/xournalpp`. Installation is not required.

Now if you want to test a PR you need to modify the git clone command as described at the bottom of [this wiki page](https://github.com/xournalpp/xournalpp/wiki/Installing). In the present case this would be
`git clone https://github.com/rolandlo/xournalpp -b fix-page-layout-update`

**Comment by LittleHuba** [comment]
> In case you have just not understood what I meant, I have no Idea how one would check out how Pull request number so and so impacts some bug I found. Could you have the grace to help me with that?

Then I need to apologize as I have definitely missunderstood you. It sounded like you are not interested in helping with debugging and just wanted to drop the ticket here. That is unfortunately more common than one would expect and makes our time just way harder than necessary. I do hope that your problem is already fixed with #2425 and should it not be the case we are sure to find the cause and provide a fix.

**Comment by pindakaas42** [comment]
> @pindakaas42
> Let me help you with how to test out a Pull Request (short: PR)
> First you need to be able to build Xournal++ from source. For Linux this is described on [the Linux build instruction wiki page](https://github.com/xournalpp/xournalpp/blob/master/readme/LinuxBuild.md).
> In short: you need to install the required dependencies, clone the repository, compile with the `cmake` tool and run from the `build` directory via the terminal command `src/xournalpp`. Installation is not required.
> 
> Now if you want to test a PR you need to modify the git clone command as described at the bottom of [this wiki page](https://github.com/xournalpp/xournalpp/wiki/Installing). In the present case this would be
> `git clone https://github.com/rolandlo/xournalpp -b fix-page-layout-update`

Will try this as soon as I have time.

**Comment by pindakaas42** [comment]
now after building from git clone https://github.com/rolandlo/xournalpp -b fix-page-layout-update: 
Xournal++ 1.1.0+dev 
└──libgtk: 3.22.30 
The bug ist still present.

**Comment by rolandlo** [comment]
@pindakaas42 
I cannot reproduce the bug as hard as I try. Do you use the hand tool by selecting it in the toolbar or via a mouse button configured to do so? Do you have the new input system enabled and the touch/scroll workaround disabled?

What do you mean exactly with this?

> This happens inconsistently, sometimes when a document is opened. But if this behaviour is seen it is consistent within that instance of xournalpp.

**Comment by pindakaas42** [comment]
Hey, new input system is activated. Mouse button and selecting it via toolbar both works.

To the consistency: I think it depends on the file I am opening, I am commenting pdfs given to me that don't belong to me, so I can't upload them here. Is there a way to open a pdf so that xournal copies the format, but remove the actual pdf, so I can upload it here?

**Comment by rolandlo** [comment]
You can simply upload the xopp-file without the pdf. Xournal++ will then ask to replace the pdf background with a different pdf file.

**Comment by pindakaas42** [comment]
I think the attached file reproduces the behaviour reliably, even a bit worse then I remember as just scrolling down then induces a little jump.
When I open this file and scroll to the second page and use the hand tool on the bottom of the first page the view jumps up to the top of the first page. 
[xournaltest.xopp.zip](https://github.com/xournalpp/xournalpp/files/5610399/xournaltest.xopp.zip)

**Comment by rolandlo** [comment]
Thanks. I think the problem is related to pages of different size. This is an unsolved problem, see #1608, and cannot be fixed with #2425. 
As long as you use documents with all pages of the same size you should not experience this problem.

**Comment by rolandlo** [comment]
Having replaced the background with this pdf
[generateBackground.zip](https://github.com/xournalpp/xournalpp/files/5610423/generateBackground.zip)
 I can reproduce the problem. Clearly this is the zooming/scrolling behavior for documents with pages of different size, which is not yet solved.

**Comment by pindakaas42** [comment]
Okay, If you say so, I did not realize it depends on the file earlier. 
I get files from different people that I comment on  in Xournal, so I can not influence this kind of stuff. But I can see that this would possibly be hard to fix for all eventualities.

**Comment by rolandlo** [comment]
As a workaround (until the scrolling/zoom behavior is fixed for pages of different size) you may consider to make all pages have the same width (e.g. scaling all of them to A4 format) with the `pdfjam` tool from the texlive distribution:
`pdfjam --outfile out.pdf --paper a4paper in.pdf`
You have to do it before you start annotating.

**Comment by pindakaas42** [comment]
Hey thanks for the the tip, I will try that
